src/gwc/src/main/java/org/geoserver/gwc/dispatch/GwcServiceDispatcherCallback.java

```diff
@@ -29,21 +36,95 @@
 public class GwcServiceDispatcherCallback extends AbstractDispatcherCallback implements
         DispatcherCallback {
 
+    private final Catalog catalog;
+
+    public GwcServiceDispatcherCallback(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
     @Override
     public Request init(Request request) {
         String context = request.getContext();
-        if (context == null || !context.startsWith("gwc/service")) {
+        if (context == null || !context.toLowerCase().contains("gwc/service")) {
```

Comment:
The virtual service workspace name will be part of the context. Let's consider this request:
`http://localhost:8080/geoserver/topp/gwc/service/wmts?REQUEST=GetCapabilities`
the context will be `topp/gwc/service`.

Regarding the good idea, what do you mean ? Using the contains to see if we are in the presence of a GWC service is to wild ?